cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1629,29 +1594,33 @@ private void handleProgressWithCompletion(ExecutionTask task, List<ExecutionTask
 
     /**
      * Periodically checks the metadata to see if leadership reassignment has finished or not.
+     * @param result Elect leaders result.
      */
-    private void waitForLeadershipTasksToFinish() {
+    private void waitForLeadershipTasksToFinish(ElectLeadersResult result) {
       List<ExecutionTask> finishedTasks = new ArrayList<>();
       Set<Long> stoppedTaskIds = new HashSet<>();
       Set<Long> deletedTaskIds = new HashSet<>();
       Set<Long> deadTaskIds = new HashSet<>();
+
+      // Process result to ensure acceptance of leader election request on broker-side and identify deleted tasks.
+      Set<TopicPartition> deletedUponSubmission = new HashSet<>();
+      ExecutionUtils.processElectLeadersResult(result, deletedUponSubmission);
+
+      boolean retry;
       do {
-        // If there is no finished tasks, we need to check if anything is blocked.
-        maybeReexecuteLeadershipTasks();
         Cluster cluster = getClusterForExecutionProgressCheck();
 
         List<ExecutionTask> slowTasksToReport = new ArrayList<>();
         boolean shouldReportSlowTasks = _time.milliseconds() - _lastSlowTaskReportingTimeMs > _slowTaskAlertingBackoffTimeMs;
         for (ExecutionTask task : inExecutionTasks()) {
           TopicPartition tp = task.proposal().topicPartition();
-          if (_stopSignal.get() == FORCE_STOP_EXECUTION) {
-            // If the execution is force-stopped, the executor will mark all in progress tasks as dead, cleanup the
-            // relevant zNodes, and then bounce the controller (see ExecutionUtils#deleteZNodesToForceStopLeadershipMoves).
-            LOG.debug("Task {} is marked as dead to force-stop the execution with a controller bounce.", task);
+          if (_stopSignal.get() != NO_STOP_EXECUTION) {
+            // If the execution is stopped, the executor will mark all in progress tasks as dead
+            LOG.debug("Task {} is marked as dead to stop the execution.", task);
             finishedTasks.add(task);
             stoppedTaskIds.add(task.executionId());
             _executionTaskManager.markTaskDead(task);
```

Comment:
So now even a regular `STOP` task will be marked as dead from now on, which is a different behavior. 
Just curious, would it 
  1. affect some other logic related to `_executionTaskManager` by surprise? 
  2. be better to remove this line?
  3. be better to `markTaskAborting()` instead?

Reply:
1. It may cause (1) the executor state to report some dead tasks and (2) relevant JMX sensor to report some dead leadership movement tasks. This has no other implication other than reporting purposes.
2. Please see the previous comment regarding pros/cons of removing it.
3. `ABORTING` is a transient state followed by `ABORTED` or `DEAD`. I don't have a strong opinion on this. We can mark it as `ABORTING` -> `ABORTED` or keep it as is. Let me know which one you'd prefer.

Reply:
Got it. Let's keep the PR simple and leave it as is. Thanks for the explanations, Efe!